include/swift/ABI/Metadata.h

```diff
@@ -3345,8 +3408,11 @@ class TargetTypeContextDescriptor
   const TargetForeignMetadataInitialization<Runtime> &
   getForeignMetadataInitialization() const;
 
+  const TargetInPlaceValueMetadataInitialization<Runtime> &
+  getInPlaceMetadataInitialization() const;
```

Comment:
I think we can put the pattern in the class context descriptor, and use a trailing object for it so it’s only there for resilient classes.

Reply:
... or just say that IncompleteMetadata is either a metadata or a pattern (instead of metadata or function). That was sort of my plan here. I wasn’t going to add an always-null field.